federation/pkg/federatedtypes/adapter.go

```diff
@@ -56,6 +56,14 @@ type FederatedTypeAdapter interface {
 	NewTestObject(namespace string) pkgruntime.Object
 }
 
+// SchedulerAdapter defines additional operations for those controllers which need
+// additional scheduling logic.
+type SchedulerAdapter interface {
```

Comment:
I don't think it's necessary to embed FederatedTypeAdapter.  I think it's preferable to keep the interfaces separate and cast a type to the Scheduler interface only where its methods are required.